app/bundles/LeadBundle/Helper/TokenHelper.php

```diff
@@ -39,9 +39,7 @@ public static function findLeadTokens($content, $lead, $replace = false)
         }
 
         // Search for bracket or bracket encoded
-        // @deprecated BC support for leadfield
         $tokenRegex = [
-            '/({|%7B)leadfield=(.*?)(}|%7D)/',
```

Comment:
Foreach below is not needed for array with one value. You could use simple variable or directly argument of `preg_match_all` function.

Reply:
Refactored in https://github.com/mautic/mautic/pull/8137/commits/ae3da9a9d4477d8d571c2241e0beccc43aad8873